core/arch/arm/plat-stm32mp1/plat_huk.c

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Linaro Limited
+ * Copyright (c) 2022, Foundries.io Limited
+ */
+
+#include <assert.h>
+#include <config.h>
+#include <crypto/crypto.h>
+#include <drivers/stm32_bsec.h>
+#include <kernel/tee_common_otp.h>
+#include <platform_config.h>
+#include <string.h>
+
+#if !defined(CFG_STM32MP1_HUK_BSEC_BASE)
+#error "CFG_STM32MP1_HUK_BSEC_BASE undefined"
+#endif
+
+static struct {
+	uint8_t key[HW_UNIQUE_KEY_LENGTH];
+	bool ready;
+} huk;
```

Comment:
um all other platforms tend to do this. I always worked under the impression that the memory would be secured. Is this not the case?

Reply:
sure, I suppose reading shadow memory and encrypting 16 bytes doesn't take long. and since the huk is only used to derive other keys performance shouldn't be that critical anyway (getting the HUK cant not be in any realtime path)